otter/test/integration/test_lib.py

```diff
@@ -0,0 +1,137 @@
+"""
+Tests for the lib functions for convergence black-box testing.
+"""
+
+from twisted.internet.defer import Deferred
+from twisted.trial.unittest import SynchronousTestCase
+
+from otter.test.integration import lib
+
+
+class IdentityV2Tests(SynchronousTestCase):
+    """Tests for the :class:`IdentityV2` class."""
+
+    def test_missing_params(self):
+        """Missing parameters to the IdentityV2 constructor should raise
+        a ValueError exception.
+        """
+        def x(u, p, e):
+            self.assertRaises(ValueError, lib.IdentityV2, None, u, p, e)
+        x("", "pw", "ep")
+        x("un", "", "ep")
+        x("un", "pw", "")
+        x(None, "pw", "ep")
+        x("un", None, "ep")
+        x("un", "pw", None)
+
+    def test_pool_kwarg(self):
+        """The pool keyword argument should be passed through to the
+        authenticate_user Otter function.
+        """
+        rcs = lib.TestResources()
+
+        class Stub(object):
+            def __init__(self):
+                self.pool = None
+
+            def authenticate_user(self, endpt, user, passwd, pool=None):
+                self.pool = pool
+                return Deferred().addCallback(lambda: {})
+
+        stub = Stub()
+        lib.IdentityV2(
+            auth=stub, username="username",
+            password="password", endpoint="endpoint"
+        ).authenticate_user(rcs)
```

Comment:
Thanks for the bug catch.  Yes, they should be called back.  The intent is to exercise the complete (albeit simplified) control flow to make sure that the pool doesn't get distorted.  I did need to call authenticate_user() though, because that's the only way stub.authenticate_user() gets invoked, and thus, the only (black-box) means of discovering what my pool setting was supplied as.